--- ./xvmisc.c.osr5	2005-05-09 07:05:16.000000000 +0000
+++ ./xvmisc.c	2005-05-09 07:05:34.000000000 +0000
@@ -49,6 +49,10 @@
 #include "bits/fc_right"
 #include "bits/fc_rightm"
 
+#if defined(sco)
+#include <sys/signal.h>
+#endif
+
 static void set_cursors PARM((Cursor, Cursor));
 
 static Atom atom_DELWIN = 0;
--- ./Makefile.osr5	2005-05-09 07:05:16.000000000 +0000
+++ ./Makefile	2005-05-09 07:05:16.000000000 +0000
@@ -2,7 +2,7 @@
 
 # your C compiler (and options) of choice
 #CC = cc
-CC = gcc -ansi
+CC = cc
 
 # use this if you're using 'cc' on a DEC Alpha (OSF/1) or MIPS (Ultrix) system:
 # CC = cc -std1 -Olimit 750
@@ -123,14 +123,14 @@
 
 
 ### for LINUX, uncomment the following line
-MCHN = -DLINUX
+#MCHN = -DLINUX
 
 
 # For SCO 1.1 (UNIX 3.2v2) machines, uncomment the following:
 #MCHN = -Dsco -DPOSIX
 #
 # For ODT 2.0 (UNIX 3.2v4) machines, uncomment the following:
-#MCHN= -Dsco -DPOSIX -DNO_RANDOM 
+MCHN= -DSVR4 -D_PTH_PTHREAD_H_ -Dsco -DPOSIX -DNO_RANDOM -I/usr/X11R6/include
 #
 # Also, you should add '-lc -lx' to the end of the LIBS def below
 # -lx must be after -lc so you get the right directory routines.
@@ -209,7 +209,7 @@
 	$(NODIRENT) $(VPRINTF) $(TIMERS) $(UNIX) $(BSDTYPES) $(RAND) \
 	$(DXWM) $(MCHN) $(PNG) $(PNGINC) $(ZLIBINC)
 
-LIBS = -lX11 $(JPEGLIB) $(TIFFLIB) $(PNGLIB) $(ZLIBLIB) -lm
+LIBS = -lX11 $(JPEGLIB) $(TIFFLIB) $(PNGLIB) $(ZLIBLIB) -lm -lsocket
 
 OBJS = 	xv.o xvevent.o xvroot.o xvmisc.o xvimage.o xvcolor.o xvsmooth.o \
 	xv24to8.o xvgif.o xvpm.o xvinfo.o xvctrl.o xvscrl.o xvalg.o \
